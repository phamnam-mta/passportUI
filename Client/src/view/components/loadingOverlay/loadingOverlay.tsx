import React from "react";
import { Spinner, SpinnerSize } from "@fluentui/react";

import "./loadingOverlay.scss";

interface ILoadingOverlayProps {
    message?: string;
}

export const LoadingOverlay = ({ message }: ILoadingOverlayProps) => {
    return (
        <div className="loading-overlay">
            <Spinner
                size={SpinnerSize.large}
                label={message}
                labelPosition="top"
                styles={{ label: { fontSize: 16, padding: 20 } }}
            />
        </div>
    );
};

export default LoadingOverlay;
