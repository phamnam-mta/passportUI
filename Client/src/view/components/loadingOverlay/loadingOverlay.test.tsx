import * as React from "react";
import { shallow } from "enzyme";
import { LoadingOverlay } from "./loadingOverlay";

describe("<LoadingOverlay />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            message: "mock-loading-text",
            location: {
                pathname: "mock/location/pathname",
            },
        };
    });

    describe("Rendering", () => {
        it("should render correctly", () => {
            const wrapper = shallow(<LoadingOverlay {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
