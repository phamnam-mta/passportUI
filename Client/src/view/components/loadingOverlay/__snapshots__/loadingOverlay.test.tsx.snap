// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingOverlay /> Rendering should render correctly 1`] = `
<div
  className="loading-overlay"
>
  <StyledSpinnerBase
    label="mock-loading-text"
    labelPosition="top"
    size={3}
    styles={
      Object {
        "label": Object {
          "fontSize": 16,
          "padding": 20,
        },
      }
    }
  />
</div>
`;
